Eastern-Research-Group/mywaterway
# Feature/432 show all layers in list

## Related Issues:
* [HMW-432](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-432)

## Main Changes:
* Switched to showing all layers (that don't have a list mode of "hide") in the layer list on every tab of the _Community_ page, rather than selectively removing some.
* Moved all layers, layer visibility state, and layer error state from the `locationSearch` context into the `Layers` context.
  * There are a good amount of keys in the context that are unused. The `errored` state object was initially going to be used in more places, but we decided not to update it based on service calls. However, some layers do still use it, and I don't think it hurts to leave the keys in place in case we want to use them in the future.
* Stopped checking for service errors when determining layer visibility. If there is a service error, the layer will simply not have any features.
* Fixed a dependency bug in the Surrounding Features widget, where changing a different surrounding layer's visibility would cause a layer's service call to abort.

## Steps To Test:
1. Navigate around the various tabs of the _Community_ page and verify that layers are not removed from the layer list or the Surrounding Features widget as tabs are changed.
2. The default layer visibility should be the same as in production for each tab.
3. One exception to the layer list inclusion is on the _Identified Issues_ tab: since the `issuesLayer` only displays a subset of the features in the `waterbodyLayer`, only the former is shown on this tab.
